src/util/services_utils.js

```diff
@@ -0,0 +1,23 @@
+/* Copyright (C) 2024 NooBaa */
+'use strict';
+
+const signature_utils = require('./signature_utils');
+const { RpcError } = require('../rpc');
+
+const sdk_type_enum = {
+    OBJECT_SDK: 'object_sdk',
+    STS_SDK: 'sts_sdk',
+    ACCOUNT_SDK: 'account_sdk',
+};
+
+function authorize_request_account_by_token(token, requesting_account) {
+    const signature_secret = token.temp_secret_key || requesting_account?.access_keys?.[0]?.secret_key?.unwrap();
+    if (signature_secret) {
```

Comment:
this condition allows unauthenticated access. I see this is not allowed for iam and sts, only for s3 because s3 also checks bucket policy. So lets add a flag to this function if to allow anonymous requests, and only send true from s3...

Reply:
Thanks, I added a boolean and changed the condition for this.